src/operator/image/image_random-inl.h

```diff
@@ -144,6 +151,45 @@ static void Normalize(const nnvm::NodeAttrs &attrs,
   });
 }
 
+inline static int FlipIndex(int idx, const int stride, const int trailing) {
+  const int low = idx % trailing;
+  int high = idx / trailing;
+  const int x = high % stride;
+  high /= stride;
+
+  return (high * stride + stride - 1 - x) * trailing + low;
+}
+
+template<typename DType>
+static void FlipImpl(const int size, DType *src, DType *dst,
+                     const int stride, const int trailing) {
+  for (int idx = 0; idx < size; ++idx) {
+    int new_idx = FlipIndex(idx, stride, trailing);
```

Comment:
Doing this on every index is slow. Either use two loops explicitly or use `inc`, see broadcasting for example

Reply:
how's it now?